## Importing necessary libraries

In [1]:
import pandas as pd
import numpy as np
from scipy import stats
from scipy.stats import norm
from scipy.stats import chi2_contingency

In [3]:
form_data = pd.read_csv('Costomer+OrderForm.csv')     # Importing dataset
form_data

Phillippines   Indonesia       Malta       India
0     Error Free  Error Free   Defective  Error Free
1     Error Free  Error Free  Error Free   Defective
2     Error Free   Defective   Defective  Error Free
3     Error Free  Error Free  Error Free  Error Free
4     Error Free  Error Free   Defective  Error Free
..           ...         ...         ...         ...
295   Error Free  Error Free  Error Free  Error Free
296   Error Free  Error Free  Error Free  Error Free
297   Error Free  Error Free   Defective  Error Free
298   Error Free  Error Free  Error Free  Error Free
299   Error Free   Defective   Defective  Error Free

[300 rows x 4 columns]

In [5]:
form_data['Phillippines'].value_counts()

Error Free    271
Defective      29
Name: Phillippines, dtype: int64

In [6]:
form_data['Indonesia'].value_counts()

Error Free    267
Defective      33
Name: Indonesia, dtype: int64

In [7]:
form_data['Malta'].value_counts()

Error Free    269
Defective      31
Name: Malta, dtype: int64

In [8]:
form_data['India'].value_counts()

Error Free    280
Defective      20
Name: India, dtype: int64

### Assume -  

**Null Hypothesis (Ho): Independence of categorical variables (Defective % of customer order forms does not vary from centre to centre)**

**Alternate Hypothesis (Ha): Dependence of categorical variables (Defective % of customer order forms vary from centre to centre)**

In [9]:
dimensional_data = np.array([[271,267,269,280],[29,33,31,20]])       # Make a contingency table
dimensional_data

array([[271, 267, 269, 280],
       [ 29,  33,  31,  20]])

In [10]:
chi2_contingency(dimensional_data) # o/p is (Chi2 stats value, p_value, df, expected observations)

(3.858960685820355,
 0.2771020991233135,
 3,
 array([[271.75, 271.75, 271.75, 271.75],
        [ 28.25,  28.25,  28.25,  28.25]]))

### Comparing p-value with alpha(0.05)

### Inference

##### Since p_value 0.277 > 0.05

##### Null Hypothesis is accepted i.e. Independence of categorical variables 

##### Therefore, we can conclude that the defective % of customer order forms does not vary from centre to centre.

**------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------**